gs.js

```diff
@@ -14,24 +14,39 @@ const submodules = [
   'pg.provider', 'pg.cursor',
 ];
 
-let gs = null;
-const lib = {};
-submodules.forEach(name => Object.assign(lib, require('./lib/' + name)));
+const api = {};
+submodules.forEach(name => Object.assign(api, require('./lib/' + name)));
 
-class GlobalStorage extends lib.StorageProvider {
-  constructor(ids) {
+const providers = {
+  fs: api.FsProvider,
+  memory: api.MemoryProvider,
+  pg: api.PostgresProvider,
+};
+
+const SYSTEM_BITMASK_SIZE = 24;
+
+class GlobalStorage extends api.StorageProvider {
+  constructor({
+    schema,
+    serverBitmask,
+    serverSuffix,
+    systemSuffix,
+    systemBitmask,
+  }) {
     super();
-    this.memory = new lib.MemoryProvider();
+    this.memory = new api.MemoryProvider();
     this.local = null;
     this.remotes = {};
     this.active = false;
     this.offline = true;
     this.nextId = 0;
     this.categories = {};
-    this.serverBitmask = ids.serverBitmask;
-    this.serverSuffix = ids.serverSuffix;
-    this.systemSuffix = ids.systemSuffix;
-    this.systemBitmask = ids.systemBitmask;
+    this.schema = schema;
```

Comment:
This leads to a recursive dependency: `Metaschema` need `gs` instance to properly load Actions whilst `GlobalStorage` needs `Metaschema` instance for initialization.
IMHO, we should initialize schema on `open`.
@tshemsedinov @lundibundi @belochub WDYT?

Reply:
Better solution: we may remove class `GlobalStorage` from `gs.js`, move all code from those class to abstract `StorageProvider` or to utils, export factory `providerFactory` from `gs.js`, so we will not pass `gs` instance to providers constructor or to `open`, we will use `this` instead. Moreover we will remove recursive abstraction what we have now. It's not obvious that `GlobalStorage` extends `StorageProvider` and wraps multiple `StorageProvider` implementations at a same time.